datafusion/functions-table/src/generate_series.rs

```diff
@@ -197,11 +197,17 @@ impl TableFunctionImpl for GenerateSeriesFuncImpl {
         }
 
         let mut normalize_args = Vec::new();
-        for expr in exprs {
+        for (expr_indice, expr) in exprs.iter().enumerate() {
             match expr {
                 Expr::Literal(ScalarValue::Null, _) => {}
                 Expr::Literal(ScalarValue::Int64(Some(n)), _) => normalize_args.push(*n),
-                _ => return plan_err!("First argument must be an integer literal"),
+                other => {
+                    return plan_err!(
+                        "Argument #{} must be an integer literal or null value, got {:?}",
```

Comment:
Is there a better way to word this, `Argument #...` seems awkward 🤷

Reply:
I don' have any good suggestions

I do think the `{:?}` might be hard to read for complex expressions. Maybe you could change this to have the display name instead. Something like

```suggestion
                        "Argument #{} must be an integer literal or null value, got {} ({:?})", expr_index, other, other,
```